compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -381,12 +390,14 @@ pub fn compile_declarative_macro(
     let rhs_nm = Ident::new(sym::rhs, def.span);
     let tt_spec = Some(NonterminalKind::TT);
 
-    // Parse the macro_rules! invocation
-    let (macro_rules, body) = match &def.kind {
+    let extract_macro_item = || match &def.kind {
```

Comment:
we can destructure once and clone the body's tokens in of `create_parser`, which does not need to take any arguments now.